internal_ws/yktrace/src/tir.rs

```diff
@@ -82,7 +82,7 @@ impl<'a, 'm> TirTrace<'a, 'm> {
             }
 
             // If a function was annotated with `do_not_trace`, skip all instructions within it as
-            // well. FIXME: recursion.
+            // well. FIXME: recursion. what does "recursion" mean?!
```

Comment:
I think this used to be recursive so we may have just forgotten to remove this comment. @vext01?

Reply:
I thought it meant that we can't handle recursive calls to `do_not_trace` functions?

Reply:
> I thought it meant that we can't handle recursive calls to do_not_trace functions?

That would be bad, surely?!

Reply:
Ah, in that case this has been fixed recently in https://github.com/softdevteam/yk/commit/b799c78df1a79a4829ee33e0210cbb1cfa4ac9e2. So this is just a left-over comment.

Reply:
> I thought it meant that we can't handle recursive calls to `do_not_trace` functions?

I misread this before. Now I know what you mean. However, does it matter if the `do_not_trace` function is recursive? The first time we see it, we replace it with a call, and then we'll never know if it is recursive or not. I'll do some testing.

Reply:
As far as I can tell, this works already, so I think we can mark this as resolved.

Reply:
I am inclined to agree.

Reply:
Er though I can't see an update to this comment in b799c78, but maybe I'm missing something?

Reply:
Yeah, sorry, I need to remove the FIXME first before resolving this.

Reply:
Okay, this turned out to not being fixed after all: b88e9fc